ImageEditor/src/test/kotlin/org/wordpress/android/imageeditor/preview/PreviewImageViewModelTest.kt

```diff
@@ -20,8 +24,44 @@ class PreviewImageViewModelTest {
     }
 
     @Test
-    fun `image shown on start`() {
-        viewModel.start(TEST_IMAGE_URL)
-        assertThat(viewModel.loadImageFromUrl.value).isEqualTo(TEST_IMAGE_URL)
+    fun `data loading started on view create`() {
+        initViewModel()
+        assertThat(viewModel.uiState.value).isInstanceOf(ImageDataStartLoadingUiState::class.java)
     }
+
+    @Test
+    fun `progress bar shown on view create`() {
+        initViewModel()
+        assertThat(requireNotNull(viewModel.uiState.value).progressBarVisible).isEqualTo(true)
+    }
+
+    @Test
+    fun `progress bar shown on low res image load success`() {
+        initViewModel()
+        viewModel.onImageLoadSuccess(TEST_LOW_RES_IMAGE_URL)
+        assertThat(requireNotNull(viewModel.uiState.value).progressBarVisible).isEqualTo(true)
+    }
+
+    @Test
+    fun `progress bar hidden on low res image load failed`() {
```

Comment:
I'm wondering if we really want to hide the progress when the low res image fails to load? The high res image can still succeed right?